adaptdl/adaptdl/speedup.py

```diff
@@ -72,14 +73,17 @@ def __init__(self, params, grad_params=None, init_batch_size=None,
                                     init_batch_size)
         else:
             self._max_batch_size = init_batch_size
+
+        self._gradient_accumulation = gradient_accumulation
+
         if params is not None:
             self._params = Params(*params)
         else:
             self._params = None
         if params is not None and init_batch_size is not None:
-            base_step_time, _, _ = _predict_log(self._params,
-                                                np.array([1]), np.array([1]),
-                                                init_batch_size)
+            base_step_time, _, _ = _predict_log(
+                self._params, np.array([1]), np.array([1]),
+                init_batch_size, 1)
```

Comment:
It's better to let the baseline performance be the best of any configuration using 1 replica. It used to be that the only configuration on 1 replica was to use the initial batch size, because the gradient noise estimation only worked using multiple replicas. However this should not be true anymore with gradient accumulation. So the baseline performance should be set to the best (local_bsz, acc_steps) achievable on 1 replica. See Eqn 15 https://arxiv.org/pdf/2008.12260.pdf.

Reply:
Fixed